core/indexer/elasticClientCommon.go

```diff
@@ -55,9 +55,16 @@ func loadResponseBody(body io.ReadCloser, dest interface{}) error {
 
 func elasticDefaultErrorResponseHandler(res *esapi.Response) error {
 	responseBody := map[string]interface{}{}
-	decodeErr := loadResponseBody(res.Body, &responseBody)
-	if decodeErr != nil {
-		return decodeErr
+	bodyBytes, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		log.Warn("cannot read elastic response body bytes", "error", err)
+		return err
+	}
+
+	notJson := json.Unmarshal(bodyBytes, &responseBody)
+	if notJson != nil {
+		log.Warn("cannot unmarshal elastic response body to map[string]interface{}", "decode error", notJson, "body response", string(bodyBytes))
+		return ErrBackOff
```

Comment:
we should return ErrBackOff only if the err string contains something like `throttled due to too many requests`, otherwise we should return the exact same error. Failing to do this will lead in incorrect masking of returned error when we will drop the extra log.Warns.

Reply:
done